resources/migration-guide-v2-v5.md

```diff
@@ -0,0 +1,402 @@
+
+```tsx
+import {
+  PrivateKey,
+  PublicKey,
+  Args,
+  DeployHeader,
+  ExecutableDeployItem,
+  Deploy,
+  DEFAULT_DEPLOY_TTL,
+  StoredContractByHash,
+  ContractHash
+} from 'casper-js-sdk';
+
+// Install a smart contract on a Casper Network
+export const install = (
+  wasm: Uint8Array,
+  args: Args,
+  paymentAmount: string,
+  sender: PublicKey,
+  chainName: string,
+  signingKey: PrivateKey
+): Deploy => {
+  const deployHeader = DeployHeader.default();
+  deployHeader.account = sender;
+  deployHeader.chainName = chainName;
+
+  const payment = ExecutableDeployItem.standardPayment(paymentAmount);
+  const deploy = Deploy.makeDeploy(
+    deployHeader,
+    payment,
+    ExecutableDeployItem.newModuleBytes(wasm, args)
+  );
+  const signedDeploy = deploy.sign(signingKeys);
+
+  return signedDeploy;
+};
+```
+
+**Example 2: Calling a Smart Contract Entrypoint.**
+
+In this example, you can see how to call an entrypoint of a smart contract using the new `ContractHash` class. This approach removes the dependency on previous abstractions, making your code easier to manage and more explicit.
+
+Key Changes
+
+- Removed Abstraction: The Contracts abstraction has been replaced with the explicit use of ContractHash and ContractPackageHash.
+- Direct Control: Developers now have full control over contract installation and entry point calls.
+- Improved Clarity: The separation of logic enhances readability and reduces confusion.
+
+```tsx
+import {
+  PrivateKey,
+  PublicKey,
+  Args,
+  DeployHeader,
+  ExecutableDeployItem,
+  Deploy,
+  DEFAULT_DEPLOY_TTL,
+  StoredContractByHash,
+  ContractHash
+} from 'casper-js-sdk';
+
+// Call an entrypoint of a smart contract.
+export const callEntrypoint = (
+  entryPoint: string,
+  args: Args,
+  sender: PublicKey,
+  chainName: string,
+  paymentAmount: string,
+  signingKeys: PrivateKey,
+  ttl: number = DEFAULT_DEPLOY_TTL
+): Deploy => {
+  const deployHeader = DeployHeader.default();
+  deployHeader.account = sender;
+  deployHeader.chainName = chainName;
+  deployHeader.ttl = ttl;
+  deployHeader.gasPrice = 1;
+
+  const session = new ExecutableDeployItem();
+  session.storedContractByHash = new StoredContractByHash(
+    ContractHash.newContract(
+      '93d923e336b20a4c4ca14d592b60e5bd3fe330775618290104f9beb326db7ae2'
+    ),
+    entryPoint,
+    args
+  );
+
+  const payment = ExecutableDeployItem.standardPayment(paymentAmount);
+  const deploy = Deploy.makeDeploy(deployHeader, payment, session);
+  const signedDeploy = deploy.sign(signingKeys);
+
+  return signedDeploy;
+};
+```
+
+---
+
+### 9. **Introduction of Casper 2.0 Transactions**
+
+#### Overview
+
+Casper 2.0 replaces deploys with a robust transaction-based system. Key components include:
+
+- `TransactionEntryPoint`
```

Comment:
How do we want developers to work with transactions in a migration scenario? We have discussed changes to building a deploy already. We don't tell them why they may need transactions instead.
@ihor ?